src/atlas/interpolation/method/Method.cc

```diff
@@ -271,6 +387,53 @@ void Method::do_execute( const Field& src, Field& tgt ) const {
     tgt.set_dirty();
 }
 
+void Method::do_execute_adjoint( FieldSet& fieldsSource, FieldSet& fieldsTarget ) const {
+    ATLAS_TRACE( "atlas::interpolation::method::Method::do_execute_adjoint()" );
+
+    const idx_t N = fieldsSource.size();
+    ATLAS_ASSERT( N == fieldsTarget.size() );
+
+    for ( idx_t i = 0; i < fieldsSource.size(); ++i ) {
+        Method::do_execute_adjoint( fieldsSource[i], fieldsTarget[i] );
+    }
+}
+
+void Method::do_execute_adjoint(Field& src, Field& tgt ) const {
+    ATLAS_TRACE( "atlas::interpolation::method::Method::do_execute_adjoint()" );
+
+    if ( nonLinear_( src ) ) {
+        throw_NotImplemented( "Adjoint interpolation only works for interpolation schemes that are linear",
+                              Here() );
+    }
+
+    if ( not missing_.empty() ) {
+        throw_NotImplemented( "Adjoint Interpolation does not work for fields that have missing data. ",
+                              Here() );
+    }
+
+    Matrix tmp(*matrix_);
+    Matrix matrix_transpose_ = tmp.transpose();
+
+    if ( src.datatype().kind() == array::DataType::KIND_REAL64 ) {
+
+        adjoint_interpolate_field<double>( src, tgt, matrix_transpose_ );
```

Comment:
Is that not equivalent to:
```interpolate_field<double>( tgt, src, matrix_transpose_ )``` ?

Reply:
No - it isn't .  Strictly speaking the adjoint is a bit more than a transpose.  
Forward
```
y = A x
x = x
```
Adjoint
```
x = x + A^t y
y = 0
```

Reply:
In that case, your approach is OK and any other technique would complicate it too much in order to avoid just a tiny bit of code duplication.